### PR TITLE
Send payment stopping email when youngest child is turning four

### PR DESCRIPTION
**Pull request by adeel718**

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/communications/PaymentCycleNotificationHandler.java [hunk 1]
```diff
@@ -54,11 +58,22 @@ private boolean voucherEntitlementIndicatesNewChildFromPregnancy(PaymentCycle pa
 
     private void handleUpcomingBirthdayEmails(PaymentCycle paymentCycle) {
         NextPaymentCycleSummary nextPaymentCycleSummary = childDateOfBirthCalculator.getNextPaymentCycleSummary(paymentCycle);
-        if (nextPaymentCycleSummary.hasChildrenTurningFour()) {
-            upcomingBirthdayEmailHandler.sendChildTurnsFourEmail(paymentCycle, nextPaymentCycleSummary);
-        }
-        if (nextPaymentCycleSummary.hasChildrenTurningOne()) {
-            upcomingBirthdayEmailHandler.sendChildTurnsOneEmail(paymentCycle, nextPaymentCycleSummary);
+        boolean isNotPregnant = isNotPregnant(paymentCycle);
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/communications/PaymentCycleNotificationHandler.java]: I don't think this adds anything making this a local variable as its not reused and looks nearly identical, we should inline this.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/communications/PaymentCycleNotificationHandler.java [hunk 2]
```diff
@@ -54,11 +58,22 @@ private boolean voucherEntitlementIndicatesNewChildFromPregnancy(PaymentCycle pa
...
+                upcomingBirthdayEmailHandler.sendChildTurnsFourEmail(paymentCycle, nextPaymentCycleSummary);
+            }
+            if (nextPaymentCycleSummary.hasChildrenTurningOne()) {
+                upcomingBirthdayEmailHandler.sendChildTurnsOneEmail(paymentCycle, nextPaymentCycleSummary);
+            }
         }
     }
-}
+
+    private boolean isNotPregnant(PaymentCycle paymentCycle) {
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/communications/PaymentCycleNotificationHandler.java]: I think this should go in the `PregnancyEntitlementCalculator` and could have a better name than this, something like `isNotPregnantAfterCycle`.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/communications/PaymentCycleNotificationHandler.java (hunk 1 above)
Comment (YetAnotherMatt) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/communications/PaymentCycleNotificationHandler.java]: true - it's a refactoring artefact I thought we'd removed again, oops.

Comment (YetAnotherMatt) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/communications/PaymentCycleNotificationHandler.java]: Done

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/communications/PaymentCycleNotificationHandler.java (hunk 2 above)
Comment (YetAnotherMatt) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/communications/PaymentCycleNotificationHandler.java]: Good point done - but the method is `claimantIsPregnantAfterCycle` to be in keeping with the existing `claimantIsPregnantInCycle`

**Pull request merged**